### PR TITLE
feat(reservation): comment display

### PR DESCRIPTION
**Pull request by Rom1-B**
Display the comment entered here in the reservation form:

image

image

+ richtext activation

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !28083

### CODE REVIEW

**Review by cedric-anne — CHANGES_REQUESTED** [review]
The reservation form is broken. I have error when I try to submit it, and reservation is not created.

```
[2023-09-11 12:54:03] glpiphplog.NOTICE: Dropdown::getDropdownUsers() in /var/www/glpi/src/Dropdown.php line 3999
Unable to decode JSON string! Is this really JSON?  {"user":"2@8add3dfa14a3"} 
[2023-09-11 12:54:05] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "begin" in /var/www/glpi/front/reservation.form.php at line 99
  Backtrace :
  public/index.php:82                                require()
  
[2023-09-11 12:54:05] glpiphplog.NOTICE:   *** PHP Deprecated function (8192): explode(): Passing null to parameter #2 ($string) of type string is deprecated in /var/www/glpi/front/reservation.form.php at line 99
  Backtrace :
  front/reservation.form.php:99                      explode()
  public/index.php:82                                require()
  
[2023-09-11 12:54:05] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key 1 in /var/www/glpi/front/reservation.form.php at line 99
  Backtrace :
  public/index.php:82                                require()
```

**Comment by Rom1-B** [comment]
I've corrected the errors in the logs.
There's still the purge button that does nothing in the modal, but it works outside.

File: templates/components/form/reservation.html.twig [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    {{ fields.dropdownField(
+        'User',
+        'users_id',
+        item.fields['users_id'],
+        __('By'),
+        {
+            entity      : item.fields['entities_id'],
+            right       : 'all'
+        }
+    ) }}
```
Comment (cedric-anne) [inline templates/components/form/reservation.html.twig:82]: 1. On new item creation, value should fallback to current connected user (code was `$uid = (empty($ID) ? Session::getLoginUserID() : $resa->fields['users_id']);`).
2. Field should be readonly in user has no right to update the reservation (there was a condition on `!Session::haveRight("reservation", UPDATE) || !Session::haveAccessToEntity($entities_id)`).

File: src/ReservationItem.php [hunk 2]
```diff
@@ -401,26 +402,13 @@ public function showForm($ID, array $options = [])
...
-
-            echo "<tr class='tab_bg_2'><td colspan='2' class='top center'>";
-            echo "<input type='submit' name='update' value=\"" . _sx('button', 'Save') . "\" class='btn btn-primary'>";
-            echo "</td></tr>\n";
-
-            echo "</table>";
-            Html::closeForm();
-            echo "</div>";
+            $options['candel'] = false;
+            $this->initForm($ID, $options);
```
Comment (cedric-anne) [inline src/ReservationItem.php:406]: This page is really hard to find (not your fault).

Display should be reworked. Previous state lacked polish, but it's worse now.

Previous state:
image

Now:
image

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**